RestaurantsNearMe/Data/Interfaces/CloudKitServiceProviding.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  CloudKitServiceProviding.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-27.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+protocol CloudKitServiceProviding {
```

Comment:
If you're abstracting clients like this with protocols (which is great!) consider using [Sourcery](https://github.com/krzysztofzablocki/Sourcery) to generate mock versions for testing.

Reply:
Still trying to figure out the best way to integrate :'(. Turns out cocoa pods wasn't linking well with my previews on Xcode 15, so i'll try the binary